geomesa-accumulo/geomesa-accumulo-datastore/src/test/scala/org/locationtech/geomesa/process/tube/TubeSelectProcessTest.scala

```diff
@@ -259,8 +262,141 @@ class TubeSelectProcessTest extends Specification {
 
       results.size mustEqual 20
     }
+
+    "should properly handle geometries crossing the IDL" in {
+      val sftName = "tubeTestType"
+      val sft = SimpleFeatureTypes.createType(sftName, s"type:String,$geotimeAttributes")
+
+      val ds = createStore
+
+      ds.createSchema(sft)
+      val fs = ds.getFeatureSource(sftName)
+
+      val featureCollection = new DefaultFeatureCollection(sftName, sft)
+
+      List("c", "d").foreach { name =>
+        List(1, 2, 3, 4).zip(List(-178, -179, 179, 178)).foreach { case (i, lat) =>
+          val sf = AvroSimpleFeatureFactory.buildAvroFeature(sft, List(), name + i.toString)
+          sf.setDefaultGeometry(WKTUtils.read(f"POINT($lat%d 0)"))
+          sf.setAttribute(DefaultDtgField, new DateTime(f"2011-01-01T00:0$i%d:00Z", DateTimeZone.UTC).toDate)
+          sf.setAttribute("type", name)
+          sf.getUserData()(Hints.USE_PROVIDED_FID) = java.lang.Boolean.TRUE
+          featureCollection.add(sf)
+        }
+      }
+      //Add a point that would be in the buffer if it wrapped around the earth
+      //This point falls between the time of C2 & C3 but west of C2. This point
+      //Would be grabbed if the buffer wrapped the globe due to the IDL.
+      val sf = AvroSimpleFeatureFactory.buildAvroFeature(sft, List(), "d5")
+      sf.setDefaultGeometry(WKTUtils.read(f"POINT(-178.5 0)"))
+      sf.setAttribute(DefaultDtgField, new DateTime(f"2011-01-01T00:02:03Z", DateTimeZone.UTC).toDate)
+      sf.setAttribute("type", "d")
+      sf.getUserData()(Hints.USE_PROVIDED_FID) = java.lang.Boolean.TRUE
+      featureCollection.add(sf)
+
+      // write the feature to the store
+      val res = fs.addFeatures(featureCollection)
+
+      // tube features
+      val tubeFeatures = fs.getFeatures(CQL.toFilter("type = 'c'"))
+
+      // result set to tube on
+      val features = fs.getFeatures(CQL.toFilter("type <> 'c'"))
+
+      // get back type b from tube
```

Comment:
[picky] Does the comment mean `type b` or `type d`?

Reply:
Thank you for catching that. Fixed. It should be d.